src/Core/Managed/Shared/TelemetryClient.cs

```diff
@@ -428,6 +428,17 @@ public void Initialize(ITelemetry telemetry)
                 var version = LazyInitializer.EnsureInitialized(ref this.sdkVersion, this.GetSdkVersion);
                 telemetry.Context.Internal.SdkVersion = version;
             }
+
+            // set RoleInstance and NodeName to the machine name if they're not initialized yet
+            if (string.IsNullOrEmpty(telemetry.Context.Internal.NodeName))
+            {
+                telemetry.Context.Internal.NodeName = PlatformSingleton.Current.GetMachineName();
+            }
+
+            if (string.IsNullOrEmpty(telemetry.Context.Cloud.RoleInstance))
+            {
+                telemetry.Context.Cloud.RoleInstance = PlatformSingleton.Current.GetMachineName();
```

Comment:
Since you already setting roleInstance manually - you can just let our endpoint populate the nodeName from the roleInstance. No need to set two equal values